File: Net/SSOExtension/VersionOne.SDK.APIClient.SSOExtension/IVersionOneSsoConfiguration.cs
namespace VersionOne.SDK.APIClient.SSOExtension
{
    /// <summary>
    /// The configuration requirements for SSO connectivity
    /// </summary>
    public interface IVersionOneSsoConfiguration
    {
        /// <summary>
        /// URL to VersionOne instance.  Must end with a slash (/)
        /// </summary>
        string ServerUrl { get; }

        /// <summary>
        /// Valid user for VersionOne instance
        /// In SSO, if your IDP uses integrated auth, you may not need to provide this information
        /// </summary>
        string Username { get; }

        /// <summary>
        /// Password for user
        /// In SSO, if your IDP uses integrated auth, you may not need to provide this information
        /// </summary>
        string Password { get; }

        /// <summary>
        /// Indicates if the connection should use integrated authentication
        /// </summary>
        bool IntegratedAuth { get; }

        /// <summary>
        /// URL to SSO Identity Provider (IDP)
        /// </summary>
        string IdpUrl { get; }

        /// <summary>
        /// Class that will parse IDP response
        /// </summary>
        string IdpResponseParser { get; }

        /// <summary>
        /// Class that will parse SP response
        /// </summary>
        string SpResponseParser { get; }

        /// <summary>
        /// URI to be used for Authentication. This parameter is required if the integrated parameter above is set to true.
        /// </summary>
        string AuthenticationUri { get; }

    }
}

File: Java/VersionOneSSO/src/com/versionone/sso/IVersionOneSSOConfiguration.java
package com.versionone.sso;

/**
 * The configuration requirements for SSO connectivity
 * 
 * @author jerry
 *
 */
public interface IVersionOneSSOConfiguration {

	/**
	 * VersionOne Server URL
	 * @return
	 */
	public String getServerUrl();
	
	/**
	 * VersionOne Username
	 * In SSO, if your IDP uses integrated auth, you may not need to provide this information
	 * 
	 * @return
	 */
	public String getUsername();
	
	/**
	 * Password for VersionOne user
	 * In SSO, if your IDP uses integrated auth, you may not need to provide this information
	 * 
	 * @return
	 */
	public String getPassword();
	
	/**
	 * Does the server support integrated auth.
	 * In SSO, set this to True if your IDP uses integrated auth
	 * @return
	 */
	public boolean getIntegratedAuth();
	
	/**
	 * This is the URL to access the Identity Provider
	 * 
	 * @return
	 */
	public String getIdpUrl();
	
	/**
	 * Name of the class that will parse the response from the Identity Provider
	 * @return
	 */
	public String getIdpResponseParser();
	
	/**
	 * Name of hte class that will parse the response from the Service Provider
	 * @return
	 */
	public String getSpResponseParser();
	
}
